src/core/errors/errors.ts

```diff
@@ -0,0 +1,36 @@
+import { Chain, ChainName } from '../constants/chains'
+
+export class NotApplicableError extends Error {
+  constructor(message = 'Not Applicable') {
+    super(message)
+
+    Error.captureStackTrace(this, NotApplicableError)
+
+    this.name = 'NotApplicableError'
+  }
+}
+
+export class NotImplementedError extends Error {
+  constructor(message = 'Not Implemented') {
+    super(message)
+
+    Error.captureStackTrace(this, NotImplementedError)
```

Comment:
this seems like black magic, what's the reason?

Reply:
That's just to prevent this constructor call from appearing in the stack trace.

It's common place when extending `Error`, but it's not critical.